### PR TITLE
[SQL Migration] Recommended configuration corrected

### PR DESCRIPTION
**Pull request by stuti149**
This PR adds recommendation configuration value on Assessment results page.

MicrosoftTeams-image (19)

### CODE REVIEW

File: extensions/sql-migration/src/api/utils.ts [hunk 1]
```diff
@@ -1169,3 +1171,83 @@ export async function clearDropDown(dropDown: DropDownComponent): Promise<void>
...
+				const vmConfigurationPreview =
+					constants.VM_CONFIGURATION_PREVIEW(dataDisk, storageDisk, tempDb);
+
+				return [vmConfiguration, vmConfigurationPreview];
+			}
+		case MigrationTargetType.SQLDB:
+			const recommendations = model._skuEnableElastic
+				? model._skuRecommendationResults.recommendations!.elasticSqlDbRecommendationResults
+				: model._skuRecommendationResults.recommendations!.sqlDbRecommendationResults;
+			const successfulRecommendationsCount = recommendations.filter(r => r.targetSku !== null).length;
```
Comment (brian-harris) [inline extensions/sql-migration/src/api/utils.ts:1215]: can recommendations.filter().length return null and throw?  
I recommend a pattern like this:
const value = recommendations.filter()?.length ?? 0;

File: extensions/sql-migration/src/dialog/skuRecommendationResults/getAzureRecommendationDialog.ts [hunk 2]
```diff
@@ -339,7 +339,7 @@ export class GetAzureRecommendationDialog {
 				const serverName = (await getSourceConnectionProfile()).serverName;
 				const errors: string[] = [];
 				try {
-					await this.skuRecommendationPage.startCardLoading();
+					// await this.skuRecommendationPage.startCardLoading();
```
Comment (brian-harris) [inline extensions/sql-migration/src/dialog/skuRecommendationResults/getAzureRecommendationDialog.ts:342]: still needed?

File: extensions/sql-migration/src/wizard/assessmentDetailsPage/assessmentDetailsHeader.ts [hunk 3]
```diff
@@ -100,16 +103,22 @@ export class AssessmentDetailsHeader {
...
-				// TODO(stutijain): replace below value with recommended config
-				value: "10 available"
+				value: configurationValue
 			},
 			{
 				value: String(migrationStateModel?._assessedDatabaseList.length)
 			},
 			{
-				value: String(migrationStateModel._assessmentResults.databaseAssessments.filter((db) => db.issues.filter(issue => issue.appliesToMigrationTargetPlatform === migrationStateModel._targetType).length === 0).length)
+				value: String(migrationStateModel._assessmentResults.databaseAssessments.filter((db) => db.issues.filter(issue => issue.appliesToMigrationTargetPlatform === migrationStateModel._targetType && issue.issueCategory === IssueCategory.Issue).length === 0).length)
```
Comment (brian-harris) [inline extensions/sql-migration/src/wizard/assessmentDetailsPage/assessmentDetailsHeader.ts:121]: same here, I recommend handling nulls for all *.length references

File: extensions/sql-migration/src/wizard/skuRecommendation/skuRecommendationPage.ts [hunk 4]
```diff
@@ -469,60 +439,27 @@ export class SKURecommendationPage extends MigrationWizardPage {
...
-
-				if (this.hasRecommendations()) {
-					const recommendations = this.migrationStateModel._skuEnableElastic
-						? this.migrationStateModel._skuRecommendationResults.recommendations!.elasticSqlDbRecommendationResults
-						: this.migrationStateModel._skuRecommendationResults.recommendations!.sqlDbRecommendationResults;
-					const successfulRecommendationsCount = recommendations.filter(r => r.targetSku !== null).length;
-					await this._dbAssessmentCard.updateSkuRecommendation(constants.RECOMMENDATIONS_AVAILABLE(successfulRecommendationsCount));
-				}
+				if (targetConfigurations.length > 0)
+					await this._dbAssessmentCard.updateSkuRecommendation(targetConfigurations[0]);
```
Comment (brian-harris) [inline extensions/sql-migration/src/wizard/skuRecommendation/skuRecommendationPage.ts:457]: nit:  please use "{ }"  after if blocks

Comment (stuti149) [reply extensions/sql-migration/src/wizard/skuRecommendation/skuRecommendationPage.ts:457]: done

File: extensions/sql-migration/src/wizard/assessmentDetailsPage/assessmentDetailsHeader.ts (hunk 3 above)
Comment (stuti149) [reply extensions/sql-migration/src/wizard/assessmentDetailsPage/assessmentDetailsHeader.ts:121]: done

File: extensions/sql-migration/src/dialog/skuRecommendationResults/getAzureRecommendationDialog.ts (hunk 2 above)
Comment (stuti149) [reply extensions/sql-migration/src/dialog/skuRecommendationResults/getAzureRecommendationDialog.ts]: removed Dialog

File: extensions/sql-migration/src/api/utils.ts (hunk 1 above)
Comment (stuti149) [reply extensions/sql-migration/src/api/utils.ts]: done

**Review by krritik — APPROVED** [review]

**Pull request merged**